components/compositing/compositor.rs

```diff
@@ -313,25 +311,29 @@ fn initialize_png(gl: &gl::Gl, width: usize, height: usize) -> RenderTargetInfo
     }
 }
 
-struct RenderNotifier {
+#[derive(Clone)]
+pub struct RenderNotifier {
     compositor_proxy: CompositorProxy,
 }
 
 impl RenderNotifier {
-    fn new(compositor_proxy: CompositorProxy,
-           _: Sender<ConstellationMsg>) -> RenderNotifier {
+    pub fn new(compositor_proxy: CompositorProxy) -> RenderNotifier {
         RenderNotifier {
             compositor_proxy: compositor_proxy,
         }
     }
 }
 
 impl webrender_api::RenderNotifier for RenderNotifier {
-    fn new_frame_ready(&mut self) {
+    fn clone(&self) -> Box<webrender_api::RenderNotifier> {
+        Box::new(RenderNotifier::new(self.compositor_proxy.clone()))
```

Comment:
Can this be just `Box::new(self.clone())`

Reply:
I feel like it becomes awkward calling self.clone() inside of a method named clone. I would rather leave it.